semantic/types_test.go

```diff
@@ -1,224 +1,219 @@
-								{Key: &semantic.Identifier{Name: "k"}, Value: &semantic.FloatLiteral{Value: 1}},
-							},
-						},
-					},
-				},
-			}),
-			b: MustType(&semantic.ObjectExpression{
-				Properties: []*semantic.Property{
-					{
-						Key: &semantic.Identifier{Name: "x"},
-						Value: &semantic.ObjectExpression{
-							Properties: []*semantic.Property{
-								{Key: &semantic.Identifier{Name: "m"}, Value: &semantic.IntegerLiteral{Value: 3}},
-								{Key: &semantic.Identifier{Name: "n"}, Value: &semantic.FloatLiteral{Value: 3}},
-							},
-						},
-					},
-					{
-						Key: &semantic.Identifier{Name: "y"},
-						Value: &semantic.ObjectExpression{
-							Properties: []*semantic.Property{
-								{Key: &semantic.Identifier{Name: "j"}, Value: &semantic.IntegerLiteral{Value: 4}},
-								{Key: &semantic.Identifier{Name: "k"}, Value: &semantic.FloatLiteral{Value: 4}},
-							},
-						},
-					},
-				},
-			}),
-			want: true,
-		},
-		{
-			name: "equal array of objects",
-			a: MustType(&semantic.ArrayExpression{
-				Elements: []semantic.Expression{&semantic.ObjectExpression{
-					Properties: []*semantic.Property{
-						{Key: &semantic.Identifier{Name: "x"}, Value: &semantic.IntegerLiteral{Value: 1}},
-						{Key: &semantic.Identifier{Name: "y"}, Value: &semantic.FloatLiteral{Value: 1}},
-					},
-				}},
-			}),
-			b: MustType(&semantic.ArrayExpression{
-				Elements: []semantic.Expression{&semantic.ObjectExpression{
-					Properties: []*semantic.Property{
-						{Key: &semantic.Identifier{Name: "x"}, Value: &semantic.IntegerLiteral{Value: 2}},
-						{Key: &semantic.Identifier{Name: "y"}, Value: &semantic.FloatLiteral{Value: 2}},
-					},
-				}},
-			}),
-			want: true,
-		},
-		{
-			name: "not equal array of objects",
-			a: MustType(&semantic.ArrayExpression{
-				Elements: []semantic.Expression{&semantic.ObjectExpression{
-					Properties: []*semantic.Property{
-						{Key: &semantic.Identifier{Name: "x"}, Value: &semantic.IntegerLiteral{Value: 1}},
-						{Key: &semantic.Identifier{Name: "y"}, Value: &semantic.FloatLiteral{Value: 1}},
-					},
-				}},
-			}),
-			b: MustType(&semantic.ArrayExpression{
-				Elements: []semantic.Expression{&semantic.ObjectExpression{
-					Properties: []*semantic.Property{
-						{Key: &semantic.Identifier{Name: "w"}, Value: &semantic.IntegerLiteral{Value: 2}},
-						{Key: &semantic.Identifier{Name: "x"}, Value: &semantic.FloatLiteral{Value: 2}},
-					},
-				}},
-			}),
-			want: false,
-		},
-	}
-	for _, tc := range testCases {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			if tc.want {
-				if tc.a != tc.b {
-					t.Errorf("expected types to be equal: pointers %p %p a: %#v b: %#v", tc.a, tc.b, tc.a, tc.b)
-				}
-			} else {
-				if tc.a == tc.b {
-					t.Errorf("expected types to not be equal: pointers %p %p a: %#v b: %#v", tc.a, tc.b, tc.a, tc.b)
-				}
-			}
-		})
-	}
-}
+// TODO(jsternberg): Readd these tests if they are still applicable.
```

Comment:
These are no longer applicable and can be deleted.  They were used to validate that the pointer magic we used to compare types was correct. That code has already been removed in the branch.